src/anet.c

```diff
@@ -59,6 +60,8 @@ static void anetSetError(char *err, const char *fmt, ...)
 
 int anetNonBlock(char *err, int fd)
 {
+    /* Depend on accept4() to  set the socket as non-blocking. */
+#ifndef HAVE_ACCEPT4
```

Comment:
This function checks the flag on the socket before turning it off, having `accept4` doesn't mean it wasn't turned off.
This function shouldn't be changed. See comment regarding `acceptCommonHandler`.